react/features/recent-list/functions.js

```diff
@@ -0,0 +1,108 @@
+/* @flow */
+
+import { RECENT_URL_STORAGE } from './constants';
+import { parseURIString } from '../base/util';
+
+
+/**
+* Retreives the recent room list and generates all the data needed
+* to be displayed.
+*
+* @param {Function} cb - The callback to be executed when the list is available.
+* @returns {void}
+*/
+export function getRecentRooms(cb: Function) {
+    window.localStorage.getItemAsync(RECENT_URL_STORAGE, recentUrls => {
+
+        if (recentUrls) {
+            const recentUrlsObj = JSON.parse(recentUrls);
+            const recentRoomDS = [];
+
+            recentUrlsObj.forEach(entry => {
+                const location = parseURIString(entry.conference);
+
+                if (location && location.room && location.hostname) {
+                    recentRoomDS.push({
+                        conference: entry.conference,
+                        dateTimeStamp: entry.date,
+                        dateString: _getDateString(entry.date),
+                        initials: _getInitials(location.room),
+                        room: location.room
+                    });
+                }
+            });
+
+            cb(recentRoomDS.reverse());
+        } else {
+            cb([]);
+        }
+    });
+}
+
+/**
+* Returns a well formatted date string to be displayed in the list.
+*
+* @param {number} dateTimeStamp - The UTC timestamp to be converted to String.
+* @returns {string}
+*/
+function _getDateString(dateTimeStamp: number) {
```

Comment:
how about using one of those nice date formatting libraries so we get some "1 minute ago" and the like?

Reply:
Good point, added moment.js, thanks!